tests/test_timeutils.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime, timedelta
+
+import pytz
+
+from ts_forecasting_pipeline.utils.time_utils import round_datetime, timedelta_fits_into, timedelta_to_pandas_freq_str, get_closest_quarter, get_most_recent_quarter
+
+
+def test_find_quarter():
+    dt = datetime(2018, 1, 26, 14, 40).astimezone(pytz.timezone("Europe/Amsterdam"))
+    recent = get_most_recent_quarter(dt)
+    assert recent.day == dt.day
+    assert recent.hour == dt.hour
+    assert recent.minute == 30
+    closest = get_closest_quarter(dt)
+    assert closest.day == dt.day
+    assert closest.hour == dt.hour
+    assert closest.minute == 45
+
+
+def test_timedelta_to_pd_freq_str():
+    assert timedelta_to_pandas_freq_str(timedelta(seconds=5)) == "5S"
+    assert timedelta_to_pandas_freq_str(timedelta(minutes=2)) == "2T"
+    assert timedelta_to_pandas_freq_str(timedelta(minutes=15)) == "15T"
+    assert timedelta_to_pandas_freq_str(timedelta(hours=1)) == "H"
+    assert timedelta_to_pandas_freq_str(timedelta(hours=2)) == "2H"
+    assert timedelta_to_pandas_freq_str(timedelta(hours=26)) == "26H"
+    assert timedelta_to_pandas_freq_str(timedelta(hours=1)) == "H"
+    assert timedelta_to_pandas_freq_str(timedelta(hours=1)) == "H"
+    assert timedelta_to_pandas_freq_str(timedelta(hours=1)) == "H"
+    assert timedelta_to_pandas_freq_str(timedelta(days=1, hours=2)) == "26H"
+    assert timedelta_to_pandas_freq_str(timedelta(days=1)) == "D"
```

Comment:
We should probably add a reference, in the docstring of this function, to the pandas user guide * about how frequency strings will be interpreted according to calendar rules, i.e. unlike timedelta objects. So the developer should make sure to apply the frequency string only to DataFrame objects with UTC datetime values.

* https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#dateoffset-objects